apps/retrospective/services/trail.go

```diff
@@ -28,3 +33,44 @@ func (service TrailService) Add(action string, actionItem string, actionItemID s
 	db.Create(&trail)
 	return
 }
+
+// GetTrails method to get history of trails for a particular sprint
+func (service TrailService) GetTrails(sprintID uint) (trails *trailSerializer.TrailSerializer, status int, err error) {
+	db := service.DB
+	trails = new(trailSerializer.TrailSerializer)
+
+	sprintTrail := db.Model(&retroModels.Trail{}).
+		Where("trails.action_item = ?", constants.ActionItemTypeMap[constants.Sprint]).
+		Where("trails.action_item_id = ?", sprintID).
+		Order("created_at DESC").QueryExpr()
+
+	sprintMemberTrail := db.Model(&retroModels.Trail{}).
+		Scopes(retroModels.TrailJoinSM).
+		Where("sprint_members.sprint_id = ?", sprintID).
+		Where("trails.action_item = ?", constants.ActionItemTypeMap[constants.SprintMember]).
+		Order("created_at DESC").QueryExpr()
+
+	sprintTaskTrail := db.Model(&retroModels.Trail{}).
+		Scopes(retroModels.TrailJoinST).
+		Where("sprint_tasks.sprint_id = ?", sprintID).
+		Where("action_item = ?", constants.ActionItemTypeMap[constants.SprintTask]).
+		Order("created_at DESC").QueryExpr()
+
+	sprintMemberTaskTrail := db.Model(&retroModels.Trail{}).
+		Scopes(retroModels.TrailJoinSMT).
+		Where("trails.action_item = ?", constants.ActionItemTypeMap[constants.SprintMemberTask]).
+		Scopes(retroModels.SprintTaskJoinSMT).
```

Comment:
We could have provided multiple scopes in the `Scopes` method, instead of providing them separately.

Also, It should be `SMTJoinSprintTask`, accordingly the order in the query might change, please check!!